### DISCUSSION TITLE
About newTestNetworkManager

### DISCUSSION THREAD

**Discussion by inancgumus** [grafana/xk6-browser#210] [Code]
@imiric 

I replaced the following code:

https://github.com/grafana/xk6-browser/blob/112193c09aa304e6a33dbdee71c10e5776faaa82/common/network_manager_test.go#L33-L65

With this one and it still works:

```golang
func newTestNetworkManager(t *testing.T, k6opts k6lib.Options) (*NetworkManager, *testSession) {
	session := &testSession{
		Session: &Session{
			id: "1234",
		},
	}

	nm := &NetworkManager{
		ctx: k6lib.WithState(
			context.Background(),
			&k6lib.State{Options: k6opts},
		),
		session: session,
		resolver: k6mockres.New(map[string][]net.IP{
			mockHostname: {
				net.ParseIP("127.0.0.10"),
				net.ParseIP("127.0.0.11"),
				net.ParseIP("127.0.0.12"),
				net.ParseIP("2001:db8::10"),
				net.ParseIP("2001:db8::11"),
				net.ParseIP("2001:db8::12"),
			},
		}, nil),
		logger: NewNullLogger(),
	}

	return nm, session
}
```

Pardon me, but I've been wondering do we need the rest of the code?

**Comment by imiric** [comment]
Probably not for the particular tests it's used in, but I think it's a good idea to leave it for completion. If a test does need those other `State` fields, I'd rather they're already there, than have to troubleshoot an obscure panic in k6 because they're missing.

So I'd prefer leaving it in, but feel free to open a PR, and in general for these cases we can dicuss it in the PR.
